src/scenarios/shared/runner.py

```diff
@@ -169,6 +179,45 @@ def run(self):
                                   ) 
             startup.runtests(self.traits)
            
+        elif self.testtype == const.CROSSGEN2:
+            startup = StartupWrapper()
+            if self.coreroot is not None and not os.path.isdir(self.coreroot):
+                getLogger().error('Cannot find CORE_ROOT at %s', self.coreroot)
+                sys.exit(1)
+            if bool(self.singlefile) == bool(self.compositefile):
+                getLogger().error("Please specify either --single <single assembly name> or --composite <absolute path of rsp file>")
+                sys.exit(1)
+        
+            compiletype = const.CROSSGEN2_COMPOSITE if self.compositefile else const.CROSSGEN2_SINGLEFILE
+            crossgen2exe = 'crossgen2%s' % extension()
+
+
+            if compiletype == const.CROSSGEN2_SINGLEFILE:
+                referencefilenames = ['System.*.dll', 'Microsoft.*.dll', 'netstandard.dll', 'mscorlib.dll']
+                referencefiles = [os.path.join(self.coreroot, filename) for filename in referencefilenames]
+                # single assembly filename: example.dll
+                filename, ext = os.path.splitext(self.singlefile)
+                outputfile = os.path.join(self.coreroot, os.path.join('out', filename+'.ni'+ext ))
+                crossgen2args = '%s -o %s -O %s' % (os.path.join(self.coreroot, self.singlefile), outputfile, ' -r '.join(['']+referencefiles))
+            
+            elif compiletype == const.CROSSGEN2_COMPOSITE:
+                # composite rsp filename: ..\example.dll.rsp
+                dllname, ext = os.path.splitext(os.path.basename(self.compositefile))
```

Comment:
use a discard here since you're not using the extension? 
```suggestion
                dllname, _= os.path.splitext(os.path.basename(self.compositefile))
```